### ISSUE TITLE
Datastore collector

### ISSUE THREAD

**Issue by cukal** [pryorda/vmware_exporter#29]
On ESXi 6.7 when enabling Datastore collector it throws:

```[2018-10-20 12:04:47.933479+00:00] Starting web server on port 9272
[2018-10-20 12:04:50.253239+00:00] Start collecting metrics from 192.168.200.234
[2018-10-20 12:04:50.297318+00:00] Starting inventory collection
[2018-10-20 12:04:50.340661+00:00] Finished inventory collection
[2018-10-20 12:04:50.340700+00:00] Starting VM performance metrics collection
[2018-10-20 12:04:50.469131+00:00] Total Virtual Machines: 4
[2018-10-20 12:04:50.471253+00:00] Finished VM performance metrics collection
[2018-10-20 12:04:50.471350+00:00] Starting VM Guests metrics collection
[2018-10-20 12:04:50.564639+00:00] Total Virtual Machines: 4
[2018-10-20 12:04:50.693015+00:00] Finished VM Guests metrics collection
[2018-10-20 12:04:50.693258+00:00] Starting datastore metrics collection
[2018-10-20 12:04:50.740913+00:00] Finished datastore metrics collection
[2018-10-20 12:04:50.741204+00:00] Starting host metrics collection
[2018-10-20 12:04:50.810167+00:00] Finished host metrics collection
[2018-10-20 12:04:50.810702+00:00] Finished collecting metrics from 192.168.200.234
[2018-10-20 12:04:50.828556+00:00] [Failure instance: Traceback: <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'replace'
/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/base.py:1276:mainLoop
/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/base.py:902:runUntilCurrent
/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py:460:callback
/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py:568:_startRunCallbacks
--- <exception caught here> ---
/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py:654:_runCallbacks
build/bdist.linux-x86_64/egg/vmware_exporter/vmware_exporter.py:133:generate_latest_metrics
]

Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/base.py", line 1276, in mainLoop
    self.runUntilCurrent()
  File "/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/base.py", line 902, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py", line 460, in callback
    self._startRunCallbacks(result)
  File "/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py", line 568, in _startRunCallbacks
    self._runCallbacks()
--- <exception caught here> ---
  File "/usr/lib/python2.7/site-packages/Twisted-18.9.0-py2.7-linux-x86_64.egg/twisted/internet/defer.py", line 654, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "build/bdist.linux-x86_64/egg/vmware_exporter/vmware_exporter.py", line 133, in generate_latest_metrics

exceptions.AttributeError: 'NoneType' object has no attribute 'replace'
```

**Comment by pryorda** [comment]
Is this pointed directly at a vcenter server or a esx host?

**Comment by cukal** [comment]
Hi Pryorda, 

Thanks for vmware_exporter ;)

It's pointed at an esx host. 

Gr,
L

**Comment by pryorda** [comment]
Ill test against one of my hosts and see what needs to change. It primarily is engineered to work with vcenter. Try disabling the metadata and let me know how it works.

**Comment by pryorda** [comment]
Have you tried disabling the metadata to see if it helps?

**Comment by pryorda** [comment]
Can you try this https://github.com/pryorda/vmware_exporter/tree/v0.2.3?

**Comment by akrauza** [comment]
Submitted a PR (#33) to resolve this issue with ESXi

**Comment by pryorda** [comment]
+1

**Comment by pryorda** [comment]
This should be resolved with #35
